include/zephyr/drivers/adc.h

```diff
@@ -707,6 +708,19 @@ __subsystem struct adc_driver_api {
 	uint16_t ref_internal;	/* mV */
 };
 
+/** @cond INTERNAL_HIDDEN */
+#define ADC_DRIVER_API(name) STRUCT_SECTION_ITERABLE(adc_driver_api, name)
```

Comment:
could you make this generic? maybe

```c
#define DRIVER_API(type) STRUCT_SECTION_ITERABLE(_api_ ## type, name)
```

or something like that and then the check could be a macro as well so you don't need to enumerate them all